Sources/Remote Logging/Crash Logging/CrashLoggingDataProvider.swift

```diff
@@ -80,4 +84,14 @@ public extension CrashLoggingDataProvider {
         guard case .enabled(let config) = performanceTracking else { return false }
         return config.trackUserInteraction
     }
+
+    /// App hang tracking is enabled by default.
+    var enableAppHangTracking: Bool {
+        return true
+    }
+
+    /// HTTP client errors are captured by default.
+    var enableCaptureFailedRequests: Bool {
+        return true
+    }
```

Comment:
I appreciated the idea of maintaining the default Sentry behavior to avoid breaking the experience of existing apps. At the same time, making it opt-in would ensure that other apps won't experience the same clutter you noted. What do you think?

```suggestion
    /// App hang tracking is disabled by default to avoid unexpected events being tracked.
    var enableAppHangTracking: Bool {
        return false
    }

    /// HTTP client errors are disabled by default to avoid unexpected events being tracked.
    var enableCaptureFailedRequests: Bool {
        return false
    }
```

Reply:
If you accept this change, I'll publish a PSA post after releasing the new version advising folks to opt-in if they need either log.

Reply:
I think that would be great! Accepting the suggested change now.

Reply:
The WP/JP team uses app hang tracking and finds it useful. But it's imprecise. As @salimbraksa suggested (p1691161246764459-slack-C05CANZ2B6F), I think we should look for other options. Apple's [MetricKit](https://developer.apple.com/documentation/metrickit) has hang diagnostics and more.

Reply:
From [SentryOptions.m](https://github.com/getsentry/sentry-cocoa/blob/main/Sources/Sentry/SentryOptions.m)

```swift
self.enableAppHangTracking = YES;
self.enableCaptureFailedRequests = YES;
```

Both app hangs and capture failed requests are enabled by Sentry by default. With this change, they are now going to be disabled.

Reply:
`enableCaptureFailedRequests` – it sounds like a good idea to disable these for good. It should typically be the server monitoring its health, not the clients. It should reduce the amount of swizzling and some overhead on the client.